@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -404,6 +404,12 @@ export class VisibleTileSet {
                 maxElevation = Math.max(maxElevation, tile.maxElevation);
             });
         });
+        // We process buildings elevation yet (via ElevationRangeSource or either way), so correct
```

Comment:
I don't like the idea of just increasing the maxElevation by the height of the highest building b/c:
1. You always do this, even if no buildings are rendered (e.g. they are disabled in the style or we just don't render vector data but raster tiles)
2. You don't know how high the actual data is. In the style the building heights could be multiplied by two or in the case of geojson it could be an arbitrary height.

I think we have to improve the min-max elevation if there is no elevationProvider and if there is an elevationProvider the min/max height of the tile should be added.

Reply:
We do not yet have the ElevationRangeSource that samples buildings data, but yes it can be improved to use real data. Although it is not a part of this ticket/issue/PR and I would not like to spread it more.... I we do not account for building height they will be (building tops) clipped when camera closes to the ground, before my stuff there we some magic numbers used, so I assume this is still better then was. See also my previous comment.